### ISSUE TITLE
Upgrading between chart versions fails

### ISSUE THREAD

**Issue by ericdill** [dask/dask-gateway#147]
Just tried to helm upgrade to 0.5.0 and the process failed. You may want to note that in the docs that the recommended approach (at least for now) is to destroy the existing namespace and create a new one.

```
 helm upgrade --install --namespace dask-gateway --values ./config.yaml dask-gateway dask-gateway/dask-gateway --version 0.5.0
UPGRADE FAILED
Error: Deployment.apps "gateway-dask-gateway" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"gateway", "app.kubernetes.io/instance":"dask-gateway", "app.kubernetes.io/managed-by":"Tiller", "app.kubernetes.io/name":"dask-gateway", "app.kubernetes.io/version":"0.5.0", "helm.sh/chart":"dask-gateway-0.5.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && Deployment.apps "scheduler-proxy-dask-gateway" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"scheduler-proxy", "app.kubernetes.io/instance":"dask-gateway", "app.kubernetes.io/managed-by":"Tiller", "app.kubernetes.io/name":"dask-gateway", "app.kubernetes.io/version":"0.5.0", "helm.sh/chart":"dask-gateway-0.5.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && Deployment.apps "web-proxy-dask-gateway" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"web-proxy", "app.kubernetes.io/instance":"dask-gateway", "app.kubernetes.io/managed-by":"Tiller", "app.kubernetes.io/name":"dask-gateway", "app.kubernetes.io/version":"0.5.0", "helm.sh/chart":"dask-gateway-0.5.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
Error: UPGRADE FAILED: Deployment.apps "gateway-dask-gateway" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"gateway", "app.kubernetes.io/instance":"dask-gateway", "app.kubernetes.io/managed-by":"Tiller", "app.kubernetes.io/name":"dask-gateway", "app.kubernetes.io/version":"0.5.0", "helm.sh/chart":"dask-gateway-0.5.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && Deployment.apps "scheduler-proxy-dask-gateway" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"scheduler-proxy", "app.kubernetes.io/instance":"dask-gateway", "app.kubernetes.io/managed-by":"Tiller", "app.kubernetes.io/name":"dask-gateway", "app.kubernetes.io/version":"0.5.0", "helm.sh/chart":"dask-gateway-0.5.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable && Deployment.apps "web-proxy-dask-gateway" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"web-proxy", "app.kubernetes.io/instance":"dask-gateway", "app.kubernetes.io/managed-by":"Tiller", "app.kubernetes.io/name":"dask-gateway", "app.kubernetes.io/version":"0.5.0", "helm.sh/chart":"dask-gateway-0.5.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

and config.yaml:

```
gateway:
  proxyToken: "6a9d6384de822ec2d9572ba4103e22ab92fc51abd3873409cec16e85a2d0aa72"

  clusterManager:
    clusterStartTimeout: 600
    workerStartTimeout: 600

  auth:
    dummy:
      password: "XXXXXXXX"

  extraConfig: |
    from dask_gateway_server.options import Options, Integer, Float, String

    def option_handler(options):
        return {
            "worker_cores": options.worker_cores,
            "worker_memory": "%fG" % options.worker_memory,
            "image": options.image,
        }

    c.DaskGateway.cluster_manager_options = Options(
        Integer("worker_cores", 2, min=1, max=4, label="Worker Cores"),
        Float("worker_memory", 8, min=2, max=16, label="Worker Memory (GiB)"),
        String("image", default="daskgateway/dask-gateway:latest", label="Image"),
        handler=option_handler,)
```

**Comment by jcrist** [comment]
Upgrading the chart works fine (meaning change your `config.yaml` and then `helm upgrade ...` with the same chart version), but it seems to fail when upgrading between chart versions. I'm not sure what's going on here.

**Comment by jcrist** [comment]
This was fixed in #150. In short - `matchLabels` should only include static information, despite the recommendations in helm's documentation. Updated the chart accordingly.
